### PR TITLE
Identity.SignOut: Apply code convention on instance + synchronized + post background

### PR DESCRIPTION
**Pull request by guperrot**
Change warning to error message for cannot sign out and update wording.

Please have a look at our [guidelines for contributions](https://github.com/Microsoft/AppCenter-SDK-Android/blob/develop/CONTRIBUTING.md) and consider the following before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] Did you add unit tests?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.

## Description

Missed feedbacks last time on sign out.

### CODE REVIEW

**Comment by guperrot** [comment]
Variables always accessed in background thread do not require extra synchronization.

File: sdk/appcenter-identity/src/main/java/com/microsoft/appcenter/identity/Identity.java [hunk 1]
```diff
@@ -431,13 +425,19 @@ public void run() {
...
-        }
-        removeTokenAndAccount();
-        AppCenterLog.info(LOG_TAG, "User sign-out succeeded.");
+    private void instanceSignOut() {
+        post(new Runnable() {
+
+            @Override
+            public void run() {
+                if (mTokenStorage.getToken() == null) {
+                    AppCenterLog.error(LOG_TAG, "Cannot sign out because not signed in.");
```
Comment (jaelim-ms) [inline sdk/appcenter-identity/src/main/java/com/microsoft/appcenter/identity/Identity.java]: Seems a warning, not an error. What about *A user hasn't signed in, skip sign-out"?

Comment (guperrot) [reply sdk/appcenter-identity/src/main/java/com/microsoft/appcenter/identity/Identity.java]: It's an error, its invalid usage of the SDK to call signOut if not signed in.

Comment (guperrot) [reply sdk/appcenter-identity/src/main/java/com/microsoft/appcenter/identity/Identity.java]: Used a different wording.

**Comment by MatkovIvan** [comment]
@guperrot As I said in the comment to that PR, `mSignInDelayed` variable can be accessed from the UI thread too.

https://github.com/Microsoft/AppCenter-SDK-Android/blob/12a347d6848aa4a2edf236bc0c796e113e3697e6/sdk/appcenter-identity/src/main/java/com/microsoft/appcenter/identity/Identity.java#L233

**Comment by guperrot** [comment]
> @guperrot As I said in the comment to that PR, `mSignInDelayed` variable can be accessed from the UI thread too.
> 
> [AppCenter-SDK-Android/sdk/appcenter-identity/src/main/java/com/microsoft/appcenter/identity/Identity.java](https://github.com/Microsoft/AppCenter-SDK-Android/blob/12a347d6848aa4a2edf236bc0c796e113e3697e6/sdk/appcenter-identity/src/main/java/com/microsoft/appcenter/identity/Identity.java#L233)
> 
> Line 233 in [12a347d](/Microsoft/AppCenter-SDK-Android/commit/12a347d6848aa4a2edf236bc0c796e113e3697e6)
> 
>  if (mSignInDelayed) {

We are getting rid of that with the new bug anyway.

**Pull request merged**